src/Lecture2.hs

```diff
@@ -74,7 +84,15 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing,[])
+removeAt 1 [x] = ( Nothing,[x])
+removeAt n list
+  | n < 0 = (Nothing, list)
+  | otherwise = ( Just (list !! n), ( take n list ++ drop (n+1) list))
```

Comment:
Have a look at the standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of writing manual recursion 🔃
You can split the list in two by index and with `splitAt` the head of the second list will be exactly the element you need to return (if the list is not empty).

You only need to handle a few corner cases (like negative input index) but that's about it 👌🏻 As you correctly mentioned, you can't use `length` due to infinite lists but you don't need `length`. Just a single call to `splitAt` is enough to traverse the list either till the end or till the splitting index 🙂